cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1656,6 +1655,8 @@ private void updateLogLevels(HttpRequest request, HttpResponder responder, Strin
       responder.sendStatus(HttpResponseStatus.OK);
     } catch (JsonSyntaxException e) {
       throw new BadRequestException("Invalid JSON in body");
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException(e.getMessage());
```

Comment:
Since `LogEntry.valueOf(String level)` can throw this exception and we want to propagate it as `BadRequestException`